plugin/CMakeLists.txt

```diff
@@ -9,6 +9,56 @@ set(plugin_SRCS
     fileinfo.cpp
     plaintextmodel.cpp
     sailfishofficeplugin.cpp
+    
```

Comment:
Just a suggestion, please ignore it if too complicated. Since this uchardet has a build system itself based on cmake, would it be nicer to use it ? Like that when uchardet is updated in the future, we don't need to update our own CMakeLists files.

Maybe there is a cmake way top compile sub cmake project in a project, or it can be done in the spec file, starting building it before the sailfish office itself.

Reply:
Thank you for suggestion, but I think, that it'll be rather complicated now because uchardet's cmake builds it as separate utility with it's own libraries and we will have some troubles with applying them while building sailfish-office itself. Actually in this case it will be better to use it as a separate library as I told you earlier (IMHO).

P.S. Thank you for an attention to uchardet library, but can you review the whole code in the project?

Reply:
I'm also a little concerned about replicating the file list here. Updating the checkout gets more complicated if the list here needs to be modified too, let alone if there starts to be compilation options. Ideally it could be built as a static lib by its own build system and just linked here. Or maybe we should just have that separate lib? Though also pondering if the size increase is worth the better detection over icu, which also could get better with updates.

For another thing noticed that the checkout is not a released version but rather the current master. Is there anything needed not included in latest 0.0.6?

Reply:
> I'm also a little concerned about replicating the file list here. Updating the checkout gets more complicated if the list here needs to be modified too, let alone if there starts to be compilation options. Ideally it could be built as a static lib by its own build system and just linked here. Or maybe we should just have that separate lib? Though also pondering if the size increase is worth the better detection over icu, which also could get better with updates.
> 
What about ICU, we had bugs with non-UTF encoding for russian text. For example if we receive a txt file with cp1251 encoding by email we will not read it correctly. I've told it to you in previous PR.
IMHO it will be better used as a separate library with its own RPM and linked in a project, as I insisted previously

> For another thing noticed that the checkout is not a released version but rather the current master. Is there anything needed not included in latest 0.0.6?

what do you mean?

Reply:
> IMHO it will be better used as a separate library with its own RPM and linked in a project, as I insisted previously

If uchardet is only used here, it makes sense not to package it separately. I've done the same for various KF5 packages for Calligra (see the calligra spec file). I think you can follow the same road here:
- compile the uchardet in the `%build` section of the spec file. Use a `-DBUILD_STATIC_LIBS=ON` and a `-DBUILD_SHARED_LIBS=OFF` flags for the cmake command. Install it in a tmp directory.
- when calling cmake for sailfish-office in the spec file, set the `CMAKE_PREFIX_PATH` variable to your tmp directory, so it can find the uchardet package. Modify the CMakeLists.txt files of plugin to look for the uchardet package and link to it accordingly.

Like that, you can use "normal" include for uchardet, like `#include <uchardet/uchardet.h>`, and if once it is decided to move uchardet in a dedicated package, just the spec file will have to be modified.

If you don't see how to do this, or have issue, I can set it up for you early next week.